### PR TITLE
Fix checkstyle config, and update style to match the now enforeced style

### PR DESCRIPTION
**Pull request by dkjellin**
Our checkstyle config _tried_ to prevent import of the `NonNull` annotation but it had put the config in the wrong place. This fixes that mistake.

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -53,7 +53,10 @@
         <module name="TypeName"/>
         <module name="IllegalImport"/> <!-- defaults to sun.* packages -->
         <module name="IllegalImport">
-            <property name="illegalPkgs" value="clover, junit.framework, org.apache.commons.lang, org.slf4j, javax.annotation.Nonnull, com.google.common.collect"/>
+            <property name="illegalPkgs" value="clover, junit.framework, org.apache.commons.lang, org.slf4j, com.google.common.collect"/>
+        </module>
+        <module name="IllegalImport">
+            <property name="illegalClasses" value="javax.annotation.Nonnull"/>
```
Comment (EliSalter-atl) [inline checkstyle.xml:59]: It is possible to group this <property> element under the other <module name="IllegalImport"> entry (with clover, junit.framework, etc), I don't think we need three <module name="IllegalImport"> entries but we do still need the self closing module on line 54 as I just tested this locally and sun.* imports get accepted without it.

**Review by EliSalter-atl — APPROVED** [review]
Looks good, I had a quick look through other projects to see if the org.springframework.lang.Nonnull got imported accidentally, it didn't. So we shouldn't need to add it :)

File: checkstyle.xml (hunk 1 above)
Comment (dkjellin) [reply checkstyle.xml:59]: probably? I copied how we handle this in our main repo, I don't mind either way to be honest.

Comment (EliSalter-atl) [reply checkstyle.xml:59]: I prefer grouping them but it's a much of a muchness 🤷

**Pull request merged**